src/SFA.DAS.Payments.ServiceFabric.Core/ServiceBusBatchCommunicationListener.cs

```diff
@@ -0,0 +1,315 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml;
+using Autofac;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.ServiceBus.Core;
+using Microsoft.Azure.ServiceBus.InteropExtensions;
+using Microsoft.Azure.ServiceBus.Management;
+using Microsoft.ServiceFabric.Services.Communication.Runtime;
+using Newtonsoft.Json;
+using SFA.DAS.Payments.Application.Infrastructure.Ioc;
+using SFA.DAS.Payments.Application.Infrastructure.Logging;
+using SFA.DAS.Payments.Application.Messaging;
+using SFA.DAS.Payments.ServiceFabric.Core.Infrastructure.UnitOfWork;
+
+namespace SFA.DAS.Payments.ServiceFabric.Core
+{
+    public interface IServiceBusBatchCommunicationListener : ICommunicationListener
+    {
+        string EndpointName { get; set; }
+    }
+
+    public class ServiceBusBatchCommunicationListener : IServiceBusBatchCommunicationListener
+    {
+        public string EndpointName { get; set; }
+        
+        private readonly IPaymentLogger logger;
+        private readonly IContainerScopeFactory scopeFactory;
+        private readonly string connectionString;
+        private readonly string errorQueueName;
+        
+        private CancellationToken startingCancellationToken;
+
+        public ServiceBusBatchCommunicationListener(string connectionString, string endpointName, string errorQueueName, IPaymentLogger logger, IContainerScopeFactory scopeFactory)
+        {
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this.scopeFactory = scopeFactory ?? throw new ArgumentNullException(nameof(scopeFactory));
+            this.connectionString = connectionString ?? throw new ArgumentNullException(nameof(connectionString));
+            EndpointName = endpointName ?? throw new ArgumentNullException(nameof(endpointName));
+            this.errorQueueName = errorQueueName ?? endpointName + "-Errors";
+        }
+
+        public Task<string> OpenAsync(CancellationToken cancellationToken)
+        {
+            startingCancellationToken = cancellationToken;
+            _ = ListenForMessages(cancellationToken);
+            return Task.FromResult(EndpointName);
+        }
+
+        protected virtual async Task ListenForMessages(CancellationToken cancellationToken)
+        {
+            await EnsureQueue(EndpointName).ConfigureAwait(false);
+            await EnsureQueue(errorQueueName).ConfigureAwait(false);
+            try
+            {
+                //TODO: allow config to determine number of listeners
```

Comment:
I'm guessing this is a legacy TODO? Is there a ticket for this?

Reply:
Yep, Legacy TODO. No idea if there's a ticket for this, nor if we need to know the amount of listeners.

Git blame says this comment was written in 2020, so I would guess it's not required.

Let me know if you want me to remove these.